src/main/anovos/data_analyzer/association_evaluator.py

```diff
@@ -215,40 +214,32 @@ def variable_clustering(
     if any(x not in idf.columns for x in list_of_cols) | (len(list_of_cols) == 0):
         raise TypeError("Invalid input for Column(s)")
 
-    idf_sample = data_sample(
-        idf,
-        fraction=min(1.0, float(sample_size) / idf.count()),
-        method_type="random",
-        seed_value=0,
-    )
-    idf_sample.persist(pyspark.StorageLevel.MEMORY_AND_DISK).count()
-
+    idf.persist(pyspark.StorageLevel.MEMORY_AND_DISK).count()
```

Comment:
For `persist`, we should create an argument and let the user choose on whether to persist or not, and their preferred persist option. For reference, check out https://github.com/anovos/anovos/blob/e0b3055a33415a211911744efbacafd0f100ad38/src/main/anovos/drift_stability/drift_detector.py#L147